src/action-ros-ci.ts

```diff
@@ -167,19 +167,39 @@ async function installRosdeps(
 	# due to difficulty reading names of some non-catkin dependencies in the ros2 core
 	# see https://index.ros.org/doc/ros2/Installation/Foxy/Linux-Development-Setup/#install-dependencies-using-rosdep
 	DEBIAN_FRONTEND=noninteractive RTI_NC_LICENSE_ACCEPTED=yes rosdep install -r --from-paths $package_paths --ignore-src --rosdistro $DISTRO -y || true`;
-	fs.writeFileSync(scriptPath, scriptContent, {mode: 0o766});
+	fs.writeFileSync(scriptPath, scriptContent, { mode: 0o766 });
 
 	let exitCode = 0;
-	const options = {cwd: workspaceDir};
+	const options = { cwd: workspaceDir };
 	if (ros1Distro) {
-		exitCode += await execBashCommand(`./${scriptName} ${ros1Distro}`, "", options);
+		exitCode += await execBashCommand(
+			`./${scriptName} ${ros1Distro}`,
+			"",
+			options
+		);
 	}
 	if (ros2Distro) {
-		exitCode += await execBashCommand(`./${scriptName} ${ros2Distro}`, "", options);
+		exitCode += await execBashCommand(
+			`./${scriptName} ${ros2Distro}`,
+			"",
+			options
+		);
 	}
 	return exitCode;
 }
 
+export async function rosdepUpdateRetry(retries: number, maxTry: number) {
```

Comment:
This function should be more generic - I think it would be better, in fact, to have `execBashCommand` take an optional argument "retries" rather than building this single purpose function.

Additionally - why are there two parameters `retries` and `maxTry`? I would expect the API to be `doSomething(retries=3)` - but you are requiring the user to always call it as `doSomething(0, 3)`, the second parameter doesn't do anything, it should be an internal implementation variable.